mantidproject/mantid
# Plotting of results for fitting table workspaces

**Description of work.**
This PR adds the functionality to fit table workspaces through the fit property browser and plot a guess and the fit results.

**To test:**
1. Open a table workspace or create one using the script below.
2. Plot the workspace
3. Open fit browser and add a suitable function 
4. Plot a guess by going to `Displace -> Plot Guess`, check this looks as expected
4. Fit the function, this should plot the calculated fit and the difference. (if `plot difference` is set to false only the calculated values should be plottted). Check this looks as expected

```
#creates an table workspace with a gaussian curve and a background 
import math

tableWS = CreateEmptyTableWorkspace()

tableWS.addColumn(type="double",name="X")
tableWS.addColumn(type="double",name="Y")
tableWS.addColumn(type="double",name="E")

for i in range(0,40):
      xValue = i * 0.05
      yValue = 10.0 * math.exp(-0.5*(xValue - 1.0)**2/0.05) + 2.0
      eValue = i%2*0.05
      tableWS.addRow ( { 'X': xValue, 'Y': yValue, 'E': eValue } )

```
Fixes #26449.
Fixes #26450.

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
